gtk/src/default/gtk-4.0/_drawing.scss

```diff
@@ -9,10 +9,10 @@
 // If $outer is true, the focus ring extends outward. Otherwise, it extends inward.
 // If $within is true, use focus-within instead of focus:focus-visible
 //
-@mixin focus-ring($target: null, $width: 2px, $offset: -$width, $outer: false, $focus-state: 'focus:focus-visible', $fc: $focus_border_color) { // Yaru: fix missing buttons transition
-  // transition-property: outline, outline-width, outline-offset, outline-color;
-  // transition-duration: 300ms;
-  // animation-timing-function: ease-in-out;
+@mixin focus-ring($target: null, $width: 2px, $offset: -$width, $outer: false, $focus-state: 'focus:focus-visible', $fc: $focus_border_color) {
+  transition-property: outline, outline-width, outline-offset, outline-color;
+  transition-duration: 300ms;
+  animation-timing-function: ease-in-out;
```

Comment:
I remember commented these lines because that was disabling state transitions for buttons.
But it should be better to add a `button` selector on the transition tweak.

Reply:
This is this jumping focus ring animation. 
I check if this is still the case. I guess button transitions are more important than the focus ring transition
Apropos transitions.... The transition for checks looks weird for the new menu checks, the little asset just shakes a little bit ^^ can we limit this change for all checks outside of menus?

Reply:
As `$button_transition` target all properties, add the transition tweak for buttons will restore state transitions and keep the focus ones 😉

About menu's checks transitions, the `not()` selector only works for the direct parent, so we must disable transitions with something like this:

```scss
popover.menu {
  check,
  radio {
    transition: none;
  }
}
```

Reply:
![Peek 2021-06-07 23-28](https://user-images.githubusercontent.com/15329494/121089997-36c2ac80-c7e8-11eb-98c2-9a8da87a9fdc.gif)

Button transitions work fine, which one do you mean?

Reply:
ah btw, upstream changed backdrop to be removed but somehow, somwhere they change the opacity in the backdrop instead - this does work with adwaita but not with yaru, only the headerbars child widgets opacity changes, I doubled checked if we got all changes and I could not find any other place :thinking: very strange

Reply:
Just checked `adwaita-1-demo` with adwaita theme and only the headerbar opacity change on backdrop.
Inside the widget-factory, the backdrop opacity works everywhere.
I think something is broken with Libadwaita upstream.